### ISSUE TITLE
Ensure the Tables list is sorted after adding a new empty table

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#631]
## Description

The list of tables on the left should be sorted alphabetically, even after adding a new empty table.

## To reproduce

1. Begin with a list of tables such as `Table 0` and `Table 1`.
1. Choose "New table", then "Empty table".
1. Expect the "Tables" list to read `Table 0`, `Table 1`, `Table 2`.
1. Observe it to be `Table 0`, `Table 2`, `Table 1`.
1. Refreshing the page will reload the tables list, sorting it correctly.

## Environment

- OS: Ubuntu 20.04
- Browser: Firefox
- Browser Version: 91

## Additional context
- Blocked by backend work, see #635

**Comment by seancolsen** [comment]
From @pavish [here](https://github.com/centerofci/mathesar/pull/617#issuecomment-913970143):

> We would need to query the table endpoint, with an explicitly mentioned order. We can address this in the stability milestones.

**Comment by kgodey** [comment]
I don't think we have sorting parameters for the table on the backend yet, we'll need a separate issue for that.

**Comment by kgodey** [comment]
I created #635, marking this issue as blocked until that's complete.

**Comment by kgodey** [comment]
This is now unblocked since the backend work has been merged.
